### PR TITLE
feat(geo): Add Map to data explorer's view type dropdown

### PR DESCRIPTION
**Pull request by Palakp41**
Closes #9213

Describe your proposed changes here:
Added a view type for Map Geo and put it behind the feature flag. Right now, we are able to select Map in the dropdown but not rendering map plot from giraffe. That implementation will be in another pr.

Kapture 2021-01-07 at 12 47 07

Screen Shot 2021-01-13 at 10 46 40 AM

- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass

### CODE REVIEW

**Comment by hercules-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 56 | :red_circle:
accessibility | 65 | :yellow_circle:
best-practices | 87 | :green_circle:
seo | 80 | :yellow_circle:
pwa | null | :question:

File: src/shared/components/GeoPlot.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  const {
+    layers,
+    zoom,
+    allowPanAndZoom,
+    detectCoordinateFields,
+    mapStyle,
+  } = viewProperties
+  const tileServerConfiguration = {
+    tileServerUrl:
+      'https://api.mapbox.com/styles/v1/influxdata/ckhl79okh00o919npquotuqxp/tiles/256/{z}/{x}/{y}@2x?access_token=pk.eyJ1IjoiaW5mbHV4ZGF0YSIsImEiOiJja2hndnM0eWQwNTVtMzRwZnd5cmQxNHhoIn0.IEAjIguiM67ql4mhiD88rw',
```
Comment (hoorayimhelping) [inline src/shared/components/GeoPlot.tsx:30]: what's up with this being hardcoded here?

File: src/shared/utils/vis.ts [hunk 2]
```diff
@@ -376,3 +377,37 @@ export const getMainColumnName = (
...
+  if (typeof value !== 'string') {
+    return null
+  }
+  return value
+}
+
+export const PRECISION_TRIMMING_TABLE = [BigInt(1)]
+for (let i = 1; i < 17; i++) {
+  PRECISION_TRIMMING_TABLE[i] = PRECISION_TRIMMING_TABLE[i - 1] * BigInt(16)
+}
```
Comment (hoorayimhelping) [inline src/shared/utils/vis.ts:396]: What's up with this `for` loop? what is `17`? should `PRECISION_TRIMMING_TABLE` be initialized with the proper values when it's exported?

File: src/shared/utils/vis.ts [hunk 3]
```diff
@@ -376,3 +377,37 @@ export const getMainColumnName = (
...
+
+export const getLatLon = (table, index: number) => {
+  const cellId = getS2CellID(table, index)
+  if (cellId === null || cellId.length > 16) {
```
Comment (hoorayimhelping) [inline src/shared/utils/vis.ts:400]: what is `16` other than something that relates to `17`? what do these magic numbers mean?

File: src/shared/components/GeoPlot.tsx (hunk 1 above)
Comment (Palakp41) [reply src/shared/components/GeoPlot.tsx]: good catch, i removed it, then rebased and pushed and forgot to check

File: src/shared/components/GeoPlot.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+}) => {
+  const {
+    layers,
+    zoom,
+    allowPanAndZoom,
+    detectCoordinateFields,
+    mapStyle,
+  } = viewProperties
+  const tileServerConfiguration = {
+    tileServerUrl: 'mapboxurlhere',
```
Comment (hoorayimhelping) [inline src/shared/components/GeoPlot.tsx:28]: maybe just leave this an empty string 🤷‍♂️

**Review by hoorayimhelping — APPROVED** [review]
Great work! much cleaner!

File: src/shared/components/GeoPlot.tsx (hunk 4 above)
Comment (Palakp41) [reply src/shared/components/GeoPlot.tsx]: Updated it :)

**Pull request merged**